### ISSUE TITLE
Attached drive, "write protected" and "write through" menu items

### ISSUE THREAD

**Issue by nicolasbauw** [dirkwhoffmann/vAmiga#844]
On a machine with an attached hard drive, I'm disturbed by a behavior : the "write through" and "write protected" items does not seem to be kept after a machine close / reopen.

More disturbing, even if "write protected" is unchecked, it rechecks itself as soon as there is a write operation on the disk, and nothing is saved, even with "write through" checked.

That, added to the fact that snapshots are not persistent, brings confusion:
- saving a game with game built-in game save option saves nothing
- snapshots are not kept
- the only way to save the game is to save the whole machine.

That works, I find this is quite confusing.

**Comment by dirkwhoffmann** [comment]
Disabling write-through mode when a snapshot is loaded is actually a feature:

```c++
void
HardDrive::_didLoad()
{
    disableWriteThrough();

    // Mark all blocks as dirty
    dirty.clear(true);
}
```

I implemented it that way because I don't want the write-through instance to be deleted by accident and to avoid several corner cases. For instance, as a fixed file name is used (e.g. hd0.hdf for drive 0), write-through can only be enabled in one emulator instance at a time. Now, suppose you work in emulator instance A with write-through enabled and load in a snapshot in instance B with write-through enabled, too. We would have an immediate conflict.

**Comment by dirkwhoffmann** [comment]
The write-protection issue is a simple yet embarrassing bug: 

```c++
        if (!getFlag(FLAG_PROTECTED)) {

            // Perform the write operation
            mem.spypeek <ACCESSOR_CPU> (addr, length, data.ptr + offset);
            
            // Handle write-through mode
            if (config.writeThrough) {
                
                wtStream[objid].seekp(offset);
                wtStream[objid].write((char *)(data.ptr + offset), length);
            }
            
            setFlag(FLAG_PROTECTED, true);
        }
```

The last line should obviously read

```c++
           setFlag(FLAG_MODIFIED, true);
```

**Comment by dirkwhoffmann** [comment]
Fixed in v3.0b3
